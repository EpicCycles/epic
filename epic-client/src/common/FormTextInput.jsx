import React from "react";
import {Icon} from "semantic-ui-react";

const FormTextInput = props => (
  <div id={props.id} className={props.className + (props.error ? " error" : "")}>
    <label>{props.label}</label>
    <input
      type="text"
      autoComplete="off"
      placeholder={props.placeholder}
      title={props.title}
      onChange={event => props.onChange(event.target.value)}
      value={props.value ? props.value : ''}
    />
    {props.value &&
    <span className="clearInput">
      <Icon
        name="remove"
        size="small"
        circular
        link
        onClick={props.onClick}
      />
    </span>}
    {props.error &&
    <div id="error-message" className="error-message error">{props.error}</div>}
  </div>
);

export default FormTextInput;