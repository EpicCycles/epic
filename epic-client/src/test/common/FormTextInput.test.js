import React from 'react';
import FormTextInput from "../../common/FormTextInput";

describe("FormTextInput tests", () => {
  it('renders the form text input correctly', () => {
    const input = shallow(
      <FormTextInput id="1234" label="Test" placeholder="Please enter your loyalty number" onChange={()=>{}} />
    );
    expect(input).toMatchSnapshot();
  });

  it('renders an error when present', ()=>{
    const input = shallow(
      <FormTextInput id="5678" label="Test" placeholder="Please enter your loyalty number" onChange={()=>{}} error="Incorrect loyalty number" />
    );
    expect(input).toMatchSnapshot();
  });

  it("should call onClearEmail when the button is clicked", () => {
    const value = "test@johnlewis.co.uk";
    const onClearEmail = jest.fn();
    const input = shallow(
      <FormTextInput
        onClick={onClearEmail}
        value={value}
      />);
    input.find("Icon").simulate("click");
    expect(onClearEmail.mock.calls.length).toBe(1);
  });

  it("should call onChange when an input is entered", () => {
    const onChange = jest.fn();
    const input = shallow(
      <FormTextInput
        onChange={onChange}
      />);
    input.find("input").simulate("change", {target:{value:1234567890123456}});
    expect(onChange.mock.calls.length).toBe(1);
    expect(onChange.mock.calls[0][0]).toBe(1234567890123456);
  });
});