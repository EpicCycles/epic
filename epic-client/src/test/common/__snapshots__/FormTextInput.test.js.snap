// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInput tests renders an error when present 1`] = `
<div
  className="undefined error"
  id="5678"
>
  <label>
    Test
  </label>
  <input
    autoComplete="off"
    onChange={[Function]}
    placeholder="Please enter your loyalty number"
    type="text"
    value=""
  />
  <div
    className="error-message error"
    id="error-message"
  >
    Incorrect loyalty number
  </div>
</div>
`;

exports[`FormTextInput tests renders the form text input correctly 1`] = `
<div
  className="undefined"
  id="1234"
>
  <label>
    Test
  </label>
  <input
    autoComplete="off"
    onChange={[Function]}
    placeholder="Please enter your loyalty number"
    type="text"
    value=""
  />
</div>
`;
